libraries/freertos_plus/standard/freertos_plus_tcp/source/FreeRTOS_TCP_IP.c

```diff
@@ -2951,8 +2990,9 @@ static BaseType_t prvTCPSendSpecialPacketHelper( NetworkBufferDescriptor_t *pxNe
 	( void ) ucTCPFlags;
 #else
 	{
+		/* Map the ethernet buffer onto the TCPPacket_t struct for easy access to the fields. */
 		TCPPacket_t *pxTCPPacket = ipPOINTER_CAST( TCPPacket_t *, pxNetworkBuffer->pucEthernetBuffer );
-		const BaseType_t xSendLength = ( BaseType_t )
+		const UBaseType_t xSendLength = ( UBaseType_t )
```

Comment:
The variable `xSendLength` is passed to `prvTCPReturnPacket()`, which expects a `uint32_t`.
Change it to:
~~~c
const uint32_t ulSendLength = ( uint32_t )
~~~
and remove the `uint32_t` cast when calling `prvTCPReturnPacket()`?